python/plugins/processing/algs/qgis/RasterCalculator.py

```diff
@@ -157,15 +160,23 @@ def _cellsize(layer):
         if cellsize == 0:
             cellsize = min([_cellsize(lyr) for lyr in layersDict.values()])
 
+        # check for layers available in the model
+        layersDictCopy = layersDict.copy() # need a shallow copy because next calls invalidate iterator
+        for lyr in layersDictCopy.values():
+            expression = self.mappedNameToLayer(lyr, expression, layersDict, context)
+
```

Comment:
There's a problem here. In my tests, it only finds one layer, and maps all layer names to that object in `mappedNameToLayer`